### ISSUE TITLE
Terminal challenge caret position

### ISSUE THREAD

**Issue by benalleng** [saving-satoshi/saving-satoshi#145]
In the current terminal challenges, the caret position does not follow the user's focus even though the functionality of moving the arrow keys behaves normally visually it does not comply.

1. Go to https://savingsatoshi.com/chapters/chapter-1/genesis-2
2. Type in something in the challenge terminal
4. Move arrow keys, while you are able to now delete text from the position you moved to with the arrow keys, the caret does not move from the end to track the user's focus making it impossible to tell where your next typed character will be.

**Expected behavior**
When the user presses the left or right arrow key, the caret position should follow their focus.

**OS/Browsers**
This Problem exists on chromium and firefox browsers

**Additional context**

The caret is currently designed with css and doesn't use any real logic to determine where the user will type the next character

```
div.react-terminal-wrapper:focus-within .react-terminal-active-input:after {
  content: '▋';
  color: white;
  font-family: monospace;
  font-family: var(--space-mono-font);
  -webkit-animation: blink 1s infinite;
  animation: blink 1s infinite;
}
```
Due to the lack of browser support for caret-shape in all browsers, I theorize the only possible fix at the moment is to completely redesign the functionality of the caret in the terminal challenge.

**Comment by afterburn** [comment]
I feel implementing a custom caret is potentially a lot of work (especially to make it work reliably across browsers/devices) for a relatively small gain. Do we not want to just use a regular caret here? @GBKS

**Comment by ecurrencyhodler** [comment]
If we remove the custom caret and use the default, we can probably move this to the demo milestone. 

If we want to use the custom caret, we should probably discuss how much work it is and compare it against other features/issues.

**Comment by GBKS** [comment]
Regular caret is fine. Maybe some someone wants to do put in the effort to make this custom one work in the future, but let's put this in the "nice to have" bucket for our future selves.

**Comment by ecurrencyhodler** [comment]
Okay confirming that the solution for this specific issue is to use the regular caret. We can create a separate feature request for the custom caret.

**Comment by ecurrencyhodler** [comment]
Closed by #201. Will create a separate issue for custom caret in the future if needed.
